pylot/simulation/utils.py

```diff
@@ -619,3 +630,22 @@ def map_ground_3D_transform_to_2D(location,
         loc_2d.y >= 0 and loc_2d.y < img_height):
         return (loc_2d.x, loc_2d.y, loc_2d.z)
     return None
+
+def match_bboxes_with_traffic_lights(bboxes, traffic_lights):
+    # Match bounding boxes with traffic lights. In order to match,
+    # the bounding box must be within 20 m of the base of the traffic light
+    # in the (x,y) plane, and must be within 5 and 7 meters above the base
+    # of the traffic light. If there are multiple possibilities, take the closest.
+    result = []
+    for bbox in bboxes:
+        best_tl_idx = 0
+        best_dist = 1000000
+        for idx in range(len(traffic_lights)):
+            tl = traffic_lights[idx]
+            dist = (bbox[0].x - tl[0].x)**2 + (bbox[0].y - tl[0].y)**2
+            if dist < best_dist and bbox[0].z - tl[0].z > 5 and bbox[0].z - tl[0].z < 7:
```

Comment:
Please add a comment that explains the rationale behind this if statement (e.g., close to traffic light, and height difference between..)